clipper_admin/clipper_admin/clipper_admin.py

```diff
@@ -0,0 +1,995 @@
+            This is the processing latency between Clipper receiving a request
+            and sending a response. It does not account for network latencies
+            before a request is received or after a response is sent.
+            If Clipper cannot process a query within the latency objective,
+            the default output is returned. Therefore, it is recommended that
+            the SLO not be set aggressively low unless absolutely necessary.
+            100000 (100ms) is a good starting value, but the optimal latency objective
+            will vary depending on the application.
+
+        Raises
+        ------
+        :py:exc:`clipper.UnconnectedException`
+        :py:exc:`clipper.ClipperException`
+        """
+        if not self.connected:
+            raise UnconnectedException()
+        url = "http://{host}/admin/add_app".format(
+            host=self.cm.get_admin_addr())
+        req_json = json.dumps({
+            "name": name,
+            "input_type": input_type,
+            "default_output": default_output,
+            "latency_slo_micros": slo_micros
+        })
+        headers = {'Content-type': 'application/json'}
+        r = requests.post(url, headers=headers, data=req_json)
+        if r.status_code != requests.codes.ok:
+            msg = "Received error status code: {code} and message: {msg}".format(
+                code=r.status_code, msg=r.text)
+            logger.error(msg)
+            raise ClipperException(msg)
+
+    def link_model_to_app(self, app_name, model_name):
+        """Routes requests from the specified app to be evaluted by the specified model.
+
+        Parameters
+        ----------
+        app_name : str
+            The name of the application
+        model_name : str
+            The name of the model to link to the application
+
+        Raises
+        ------
+        :py:exc:`clipper.UnconnectedException`
+        :py:exc:`clipper.ClipperException`
+
+        Note
+        -----
+        Both the specified model and application must be registered with Clipper, and they
+        must have the same input type. If the application has previously been linked to a different
+        model, this command will fail.
+        """
+        if not self.connected:
+            raise UnconnectedException()
+
+        url = "http://{host}/admin/add_model_links".format(
+            host=self.cm.get_admin_addr())
+        req_json = json.dumps({
+            "app_name": app_name,
+            "model_names": [model_name]
+        })
+        headers = {'Content-type': 'application/json'}
+        r = requests.post(url, headers=headers, data=req_json)
+        logger.info(r.text)
+        if r.status_code != requests.codes.ok:
+            msg = "Received error status code: {code} and message: {msg}".format(
+                code=r.status_code, msg=r.text)
+            logger.error(msg)
+            raise ClipperException(msg)
+
+    def build_and_deploy_model(self,
```

Comment:
In addition to this method and `deploy_model`, I still think it's a good idea to have a `build_model` method defined within this module that performs only the first operation documented below. This will be useful for data scientists who want to deploy their models without access to (or a need for) a running Clipper instance.

Reply:
Yeah good point. Done.